src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+        remaining = table.copy().loc[~table.index.isin(original_rows)]
+        joined_rows = error_table.merge(
+            remaining,
+            how="left",
+            on=join_vars,
+            suffixes=["_original", "_duplicate"],
+        )
+        return joined_rows
+
+    def _extended_validation(
+        self, output_path: Union[str, pathlib.Path], scheme: str = "green_dark"
+    ) -> None:
+        """Generate HTML outputs of impacted rows from GTFS errors/warnings.
+
+        Parameters
+        ----------
+        output_path : Union[str, pathlib.Path]
+            The path to save the HTML output to
+        scheme : str, optional
+            Colour scheme from pretty_html_table, by default "green_dark".
+            Colour schemes can be found here:
+            https://pypi.org/project/pretty-html-table/
+
+        Returns
+        -------
+        None
+
+        """
+        table_map = {
+            "agency": self.feed.agency,
+            "routes": self.feed.routes,
+            "stop_times": self.feed.stop_times,
+            "stops": self.feed.stops,
+            "trips": self.feed.trips,
+            "calendar": self.feed.calendar,
+        }
+
+        # determine which errors/warnings have rows that can be located
+        validation_table = self.is_valid()
+        validation_table["valid_row"] = validation_table["rows"].apply(
+            lambda x: 1 if len(x) > 0 else 0
+        )
+        ext_validation_table = validation_table.copy()[
+            validation_table["valid_row"] == 1
+        ]
+        # locate the impacted rows for each error
+        for table, rows, message, msg_type in zip(
+            ext_validation_table["table"],
+            ext_validation_table["rows"],
+            ext_validation_table["message"],
+            ext_validation_table["type"],
+        ):
+            # create a more informative table for repeated pairs
+            if "Repeated pair" in message:
+                join_vars = (
+                    message.split("(")[1]
+                    .replace(")", "")
+                    .replace(" ", "")
+                    .split(",")
+                )
+                drop_cols = [
+                    col
+                    for col in GTFS_UNNEEDED_COLUMNS[table]
+                    if col not in join_vars
+                ]
+                filtered_tbl = table_map[table].copy().drop(drop_cols, axis=1)
+                impacted_rows = self._create_extended_repeated_pair_table(
+                    table=filtered_tbl,
+                    join_vars=join_vars,
+                    original_rows=rows,
+                )
+                base_columns = [
+                    item
+                    for item in list(filtered_tbl.columns)
+                    if item not in join_vars
+                ]
+                duplicate_counts = {}
+                for col in base_columns:
+                    duplicate_counts[col] = impacted_rows[
```

Comment:
Currently no tests for this behaviour. How challenging to produce this? Would we need another fixture or does the Newport fixture present a repeated pair?

Reply:
We would need to add another feature as the Newport data does not include a repeated pair.

Reply:
typo above, needs a new fixture that can reproduce the repeated pair context. I will raise an issue for this.